File: projects/Ruby/examples/serialization.rb
require_relative '../proto/proto'

# Create a new account with some orders
account = Proto::Account.new(1, 'Test', Proto::State.good, Proto::Balance.new('USD', 1000.0), Proto::Balance.new('EUR', 100.0))
account.orders.push(Proto::Order.new(1, 'EURUSD', Proto::OrderSide.buy, Proto::OrderType.market, 1.23456, 1000.0))
account.orders.push(Proto::Order.new(2, 'EURUSD', Proto::OrderSide.sell, Proto::OrderType.limit, 1.0, 100.0))
account.orders.push(Proto::Order.new(3, 'EURUSD', Proto::OrderSide.buy, Proto::OrderType.stop, 1.5, 10.0))

# Serialize the account to the FBE stream
writer = Proto::AccountModel.new(FBE::WriteBuffer.new)
writer.serialize(account)
raise "Verify error!" unless writer.verify

# Show the serialized FBE size
puts "FBE size: #{writer.buffer.size}"

# Deserialize the account from the FBE stream
reader = Proto::AccountModel.new(FBE::ReadBuffer.new)
reader.attach_buffer(writer.buffer)
raise "Verify error!" unless reader.verify
reader.deserialize(account)

# Show account content
puts
puts account

File: projects/JavaScript/examples/serialization.js
/* eslint-disable prefer-const,no-loss-of-precision */
'use strict'

const fbe = require('../proto/fbe')
const proto = require('../proto/proto')

// Create a new account with some orders
let account = new proto.Account(1, 'Test', proto.State.good, new proto.Balance('USD', 1000.0), new proto.Balance('EUR', 100.0))
account.orders.push(new proto.Order(1, 'EURUSD', proto.OrderSide.buy, proto.OrderType.market, 1.23456, 1000.0))
account.orders.push(new proto.Order(2, 'EURUSD', proto.OrderSide.sell, proto.OrderType.limit, 1.0, 100.0))
account.orders.push(new proto.Order(3, 'EURUSD', proto.OrderSide.buy, proto.OrderType.stop, 1.5, 10.0))

// Serialize the account to the FBE stream
let writer = new proto.AccountModel(new fbe.WriteBuffer())
writer.serialize(account)
console.assert(writer.verify())

// Show the serialized FBE size
console.log(`FBE size: ${writer.buffer.size}`)

// Deserialize the account from the FBE stream
let reader = new proto.AccountModel(new fbe.ReadBuffer())
reader.attachBuffer(writer.buffer)
console.assert(reader.verify())
reader.deserialize(account)

// Show account content
console.log()
console.log(account)
